src/components/app/MenuButtons/index.js

```diff
@@ -128,10 +128,22 @@ class MenuButtonsImpl extends React.PureComponent<Props, State> {
   };
 
   _onProfileDeleted = () => {
-    this.props.profileRemotelyDeleted();
     this.setState({
       metaInfoPanelState: 'profile-deleted',
     });
+    // As of React 18, If the action dispatch isn't delayed, the component will
+    // get rerendered with the redux changes before the react state changes are
+    // applied. As a result CurrentProfileUploadedInformationLoader could reset
+    // the profile information faster than this metaInfoPanelState was applied,
+    // and we were getting an error in _renderMetaInfoButton.
+    // Using requestAnimationFrame ensures that this component's state
+    // is properly applied before changing the redux state. Note that if we were
+    // using a function component, we could use useLayoutEffect instead.
+    // Note that this might not be necessary in future versions of React or
+    // React-Redux, so be sure to revisit this choice in the future.
+    requestAnimationFrame(() => {
+      this.props.profileRemotelyDeleted();
+    });
```

Comment:
Hm this is a bit hacky, but if it works it's okay for me I guess. Do we always know for sure that requestAnimationFrame will be enough for it to update the state?

Another solution might be to have a `componentDidUpdate` that checks if `metaInfoPanelState` state has changed, and calls `profileRemotelyDeleted` only if it's changed.